source/extensions/transport_sockets/tls/context_impl.h

```diff
@@ -110,6 +110,8 @@ class ContextImpl : public virtual Envoy::Ssl::Context {
   bool verifyCertChain(X509& leaf_cert, STACK_OF(X509) & intermediates, std::string& error_details);
 
 protected:
+  friend class SslContextStatsTest;
+
```

Comment:
Instead of a friend function, just use a test-only subclass of ContextImpl that exposes the protected constructor.

Reply:
Good idea; done.